spec/factorial_spec.rb

```diff
@@ -0,0 +1,35 @@
+require_relative '../tdd_project'
+
+describe 'Solver tests drivers' do
+  solver = Solver.new
+
+  describe '#factorial' do
+    it 'has a factorial method' do
+      expect(solver).to respond_to(:factorial)
+    end
+
+    it 'factorial returns a number' do
+      expect(solver.factorial(10)).to be_a(Integer)
+    end
+
+    it 'returns the factorial of the number' do
+      expect(solver.factorial(3)).to eq(6)
+    end
+
+    it 'returns 1 for 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'returns 1 for 1' do
+      expect(solver.factorial(1)).to eq(1)
+    end
+
+    it 'returns 1 for negative numbers' do
+      expect(solver.factorial(-1)).to eq(1)
+    end
```

Comment:
- [ ] As discussed above, the requirements expect that the method raises an exception for negative numbers. It should not return `1` or any other value, for negative numbers. Please work on this.